### PR TITLE
feat: upgraded to node v18, added .nvmrc and updated workflows

### PR DESCRIPTION
**Pull request by Mashal-m**
- Added support for node v18, updated .nvmrc and workflows.

**Ticket**
[Upgrade Node JS from 16 to 18](https://github.com/openedx/frontend-app-authn/issues/780)

### CODE REVIEW

**Review by abdullahwaheed — APPROVED** [review]
looks good to me!
@zainab-amir could you please take a look as well

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
   "dependencies": {
     "@edx/brand": "npm:@edx/brand-openedx@1.2.0",
     "@edx/frontend-component-cookie-policy-banner": "2.2.2",
-    "@edx/frontend-platform": "3.6.1",
+    "@edx/frontend-platform": "^4.0.2",
```
Comment (zainab-amir) [inline package.json:37]: Why restrict it to v4 only?
```suggestion
    "@edx/frontend-platform": "4.0.2",
```

**Review by zainab-amir — COMMENTED** [review]
Please squash your commits

File: package.json (hunk 1 above)
Comment (Mashal-m) [reply package.json:37]: Because we released the node18 upgrade in frontend-platform version 4.

Comment (zainab-amir) [reply package.json:37]: Adding the caret (^) will update to all future minor/patch versions, without incrementing the major version. We should not restrict it unless we have reason to do it.

Comment (Mashal-m) [reply package.json]: removed

**Comment by Mashal-m** [comment]
> Please squash your commits

Squashed.

**Pull request merged**